src/components/SearchDrawer/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { useQuery } from 'react-query';
+import { useRouter } from 'next/router';
+import {
+    Box,
+    Container,
+    Typography,
+} from '@material-ui/core';
+import { FunctionComponent, useEffect, useRef, useState } from 'react';
+
+import getSearchDrawerResults from '../../fetching/getSearchDrawerResults';
+import useDebounce from '../../hooks/useDebounce';
+
+import ResultsList from './ResultsList';
+import SearchField from './SearchField';
+
+export const MINIMUM_CHARACTERS = 3;
+
+interface SearchDrawerProps {
+    orgId: string;
+}
+
+const SearchDrawer: FunctionComponent<SearchDrawerProps> = ({ orgId }): JSX.Element | null => {
+    const router = useRouter();
+
+    const drawer = useRef<HTMLDivElement>(null);
+    const [drawerOpen, setDrawerOpen] = useState(false);
+    const [searchFieldValue, setSearchFieldValue] = useState<string>('');
+
+    const { refetch, data: results, isIdle, isFetching } = useQuery(
+        ['searchDrawerResults', searchFieldValue],
+        getSearchDrawerResults(searchFieldValue, orgId),
+        { enabled: false },
+    );
+
+    const debouncedQuery = useDebounce(async () => {
+        refetch();
+    }, 600);
+
+    // Watch for changes on the search field value and debounce search if changed
+    useEffect(() => {
+        if (searchFieldValue.length >= MINIMUM_CHARACTERS) debouncedQuery();
```

Comment:
Here you're invoking `debouncedQuery()` which based on the typing of `useDebounce()` I would have thought could be `undefined`. But obviously typescript isn't complaining, so I guess I'm misunderstanding something. How are you able to invoke `debouncedQuery()` here without checking for `undefined`?

Reply:
The typing doesn't show that the value of `debouncedQuery` would possibly be undefined, but that the value of the output of that function might possibly be undefined. I think the lodash debouncer always returns a function, but that function might return undefined instead of another function. That's why it's not showing any type errors.